pkg/providers/tinkerbell/pbnj/client.go

```diff
@@ -0,0 +1,65 @@
+package pbnj
+
+import (
+	"context"
+	"os"
+
+	"github.com/tinkerbell/cluster-api-provider-tinkerbell/pbnj/client"
+	"github.com/tinkerbell/pbnj/api/v1"
+)
+
+const (
+	PbnjGrpcAuth = "PBNJ_GRPC_AUTHORITY"
+)
+
+type Pbnj struct {
+	*client.PbnjClient
+}
+
+type BmcSecretConfig struct {
+	Host     string
+	Username string
+	Password string
+	Vendor   string
+}
+
+func NewPBNJClient(pbnjGrpcAuth string) (*Pbnj, error) {
+	os.Setenv(PbnjGrpcAuth, pbnjGrpcAuth)
+
+	conn, _ := client.SetupConnection()
+
+	mClient := v1.NewMachineClient(conn)
+	tClient := v1.NewTaskClient(conn)
+	pbnjObj := &Pbnj{client.NewPbnjClient(mClient, tClient)}
+
+	os.Unsetenv(PbnjGrpcAuth)
```

Comment:
To make it clear for maintainers this could do with being deferred immediately after the `os.Setenv()`.